modules/programs/buf.nix

```diff
@@ -0,0 +1,23 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let cfg = config.programs.buf;
+in {
+  meta.maintainers = [ maintainers.lucperkins ];
+
+  options.programs.buf = {
+    enable = mkEnableOption "Buf: a CLI tool for working with Protocol Buffers";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.buf;
+      defaultText = literalExpression "pkgs.buf";
+      description = "Package providing <command>buf</command>.";
+    };
+  };
+
+  config = mkIf cfg.enable {
```

Comment:
Not sure how much value this package is offering as is. Is there any config for this tool that we can provide options for?

Reply:
That's a fair question. There will probably be more configurability in the future but if this doesn't add much value now, I'm happy to close this and take it as a learning exercise.